.buildkite/scripts/imageprep.sh

```diff
@@ -2,23 +2,47 @@
 
 set -euo pipefail
 
-source .buildkite/scripts/setenv.sh
-
-DOCKER_REGISTRY = 'docker.elastic.co'
-DOCKER_REGISTRY_SECRET_PATH = "kv/ci-shared/platform-ingest/docker_registry_prod"
+DOCKER_REGISTRY_SECRET_PATH="kv/ci-shared/platform-ingest/docker_registry_prod"
 
 publish_docker_image() {
-    echo "Pushing the docker image "$DOCKER_IMAGE:$DOCKER_IMAGE_TAG" to the ${DOCKER_REGISTRY} registry..."
-    dockerLogin
-    docker push "$DOCKER_IMAGE:$DOCKER_IMAGE_TAG"
+    echo "Pushing the docker image "$DOCKER_IMAGE":"$DOCKER_IMAGE_TAG" to the "${DOCKER_REGISTRY}" registry..."
+    docker_login
+    docker push "${DOCKER_IMAGE}":"${DOCKER_IMAGE_TAG}"
     docker logout "$DOCKER_REGISTRY"
 }
 
-
-dockerLogin() {
-    DOCKER_USER=$(retry 5 vault kv get -field user ${DOCKER_REGISTRY_SECRET_PATH})
-    DOCKER_PASSWORD=$(retry 5 vault kv get -field password ${DOCKER_REGISTRY_SECRET_PATH})
+docker_login() {
+    DOCKER_USER=$(retry 5 vault kv get -field user "${DOCKER_REGISTRY_SECRET_PATH}")
```

Comment:
Following what it is done in `apm-pipeline-library` ([withDockerEnv](https://github.com/elastic/apm-pipeline-library/blob/bafc85a84dda5436498e1abd5ede9ee9499e6e6c/vars/withDockerEnv.groovy#L43-L45)), please set this environment variable to be redacted too. cc @alexsapran 

So the change here would be could be something like (other suffixes could be used):

```suggestion
    DOCKER_USER_SECRET=$(retry 5 vault kv get -field user "${DOCKER_REGISTRY_SECRET_PATH}")
```
In one of the latest builds ([link](https://buildkite.com/elastic/fleet-server/builds/268#01880f32-a528-4ef4-9bd5-def139ab5273)), I don't see that string (the user). But I would follow the same as done in apm-pipeline-library.

Reply:
I'm not sure I understood your question, please correct me if I'm wrong. The mentioned variables aren't global variables, I defined them in the docker_login func. It means they are using only inside the func. In addition in the mentioned URL the variables with the same name as I used.
Also, in the doc https://buildkite.com/docs/pipelines/managing-log-output#redacted-environment-variables I found _SECRET and _PASSWORD patterns as well. 
Again, perhaps I didn't get the question :)

Reply:
Even if they are declared inside the docker_login func, I would follow the same conditions to redact them. We need to put these mechanisms to ensure that they are not leaked.

And talking about that, what it has been done in some other repositories to get those secrets, it is to use the `pre-command` hook. For instance:
- kibana:
    - https://github.com/elastic/kibana/blob/main/.buildkite/hooks/pre-command 
    -  https://github.com/elastic/kibana/blob/main/.buildkite/scripts/lifecycle/pre_command.sh
- elastic-pacakge:
    - https://github.com/elastic/elastic-package/blob/main/.buildkite/hooks/pre-command
    
    
I would try to follow that pattern.

Reply:
> Should I add something else to enable the `pre-command` hook?

No, it is automatically enabled when it is detected (ensure that it has execution permission that script)